dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -939,9 +944,199 @@ internal class DdAndroidGradlePluginFunctionalTest {
+        )
+        stubNativeFiles()
+
+        val color = forge.anElementFrom(colors)
+        val version = forge.anElementFrom(versions)
+        val variantVersionName = version.lowercase()
+        val variant = "${version.lowercase()}$color"
+        val taskName = resolveNdkSymbolUploadTask(variant)
+
+        gradleRunner { withArguments("--info", ":samples:app:assembleRelease") }
+            .build()
+
+        val result = gradleRunner {
+            withArguments(
+                taskName,
+                "--info",
+                "--stacktrace",
+                "-PDD_API_KEY=fakekey"
+            )
+        }
+            .buildAndFail()
+
+        // Then
+        val buildIdInOriginFile = testProjectDir.findBuildIdInOriginFile(variant)
+
+        assertThat(result).containsInOutput("Creating request without GZIP encoding.")
+
+        assertThat(result).containsInOutput("Uploading ndk_symbol_file file: ")
+
+        assertThat(result).containsInOutput(
+            "Uploading file libplaceholder.so with tags " +
+                "`service:com.example.variants.$variantVersionName`, " +
+                "`version:1.0-$variantVersionName`, " +
+                "`version_code:1`, " +
+                "`variant:$variant`, " +
+                "`build_id:$buildIdInOriginFile` (site=datadoghq.com):"
+        )
+
+        for (arch in SUPPORTED_ABIS) {
+            assertThat(result).containsInOutput("extra attributes: {arch=$arch}")
+        }
+    }
+
+    // endregion
+
+    @Test
+    fun `M try to upload the symbol file and mapping file W upload { using a fake API_KEY }`(forge: Forge) {
+        // Given
+        stubGradleBuildFromResourceFile(
+            "native_gradle_files/build_with_proguard_native_and_datadog_dep.gradle",
+            appBuildGradleFile
+        )
+        stubNativeFiles()
+
+        val color = forge.anElementFrom(colors)
+        val version = forge.anElementFrom(versions)
+        val variantVersionName = version.lowercase()
+        val variant = "${version.lowercase()}$color"
+        val ndkSymbolUploadTaskName = resolveNdkSymbolUploadTask(variant)
+        val mappingUploadTaskName = resolveMappingUploadTask(variant)
+
+        // When
+        gradleRunner { withArguments("--info", ":samples:app:assembleRelease") }
+            .build()
+
+        val nativeResult = gradleRunner {
+            withArguments(
+                ndkSymbolUploadTaskName,
+                "--info",
+                "--stacktrace",
+                "-PDD_API_KEY=fakekey"
+            )
+        }
+            .buildAndFail()
+
+        val mappingResult = gradleRunner {
+            withArguments(
+                mappingUploadTaskName,
+                "--info",
+                "--stacktrace",
+                "-PDD_API_KEY=fakekey"
+            )
+        }
+            .buildAndFail()
+
+        // Then
+        val buildIdInOriginFile = testProjectDir.findBuildIdInOriginFile(variant)
```

Comment:
paranoidal check: let's check also that this build ID exists in the apk/aab generated as well, after you executed upload tasks (to make sure that upload tasks don't re-trigger APK/AAB generation once again).